react/features/app/actions.js

```diff
@@ -194,6 +179,15 @@ export function appWillUnmount(app) {
  * @returns {Promise<Object>}
  */
 function _loadConfig(location: Object) {
+    // Here we 'fake' a configuration when the room is undefined. We cannot
+    // load a configuration from the server, but we need lib-jitsi-meet to be
+    // initialized in order to be able to create some local tracks for the
+    // welcome page.
+
+    if (typeof location.room === 'undefined') {
+        return Promise.resolve({});
+    }
```

Comment:
I picked most pieces of this PR and either merged them in master (e.g. comments) or separated them in https://github.com/jitsi/jitsi-meet/pull/1958 (e.g. adding the `?room=` to config.js's URL). I'm still experimenting with ideas revolving around dealing with the possible delay imposed by the downloading of config.js in general so please don't spend any more time of this PR, please don't rebase or close it.

The one comment I'd like to make (well, reiterate because I remember I mentioned it when we talked but I accept I may have been brief) about downloading config.js only when a room is specified: technically, the current Web uses config and interfaceConfig on the WelcomePage because that allows deployments to customize it. I know Web will continue to work right now because it uses the global config variable, we don't "load" interfaceConfig even but I'd like to look at this from a future perspective.

Reply:
But the web and mobile are different and I'm not sure we'll be able to fold both on every account. If we load config.js also on the welcome page on mobile, we still need to fetch it when going into a room, and because the URL is different, caching won't help us AFAIS. Then there is interface_config, which, as you obviously know (but for the sake of completeness I'm stating here) we don't currently load. Iff that file is not dependent on the shard we could load it early (as we used to with config.js) and cache it. I didn't make any movement in that direction because as of right now they are separate files so I focused solely on config.js. That said, I did (still do) have some ideas on how to go about it.

Please keep me in the loop if you intend to make architectural differences at the config loading level.